### ISSUE TITLE
Remote communication not working after tt_metal runtime on galaxy

### ISSUE THREAD

**Issue by broskoTT** [tenstorrent/tt-umd#320]
Reported by @tt-vjovanovic

@pjanevskiTT and me still lack enough knowledge about how remote communication works on our chips to quickly pinpoint the exact issue.
@tt-vjovanovic suspects the culprit might be related to https://github.com/tenstorrent/tt-umd/blob/main/device/cluster.cpp#L2139C112-L2139C118 : remote_transfer_ethernet_cores[mmio_capable_chip_logical].at(0)

The symptom is that using read_from_non_mmio_device works in UMD when tt_debude resets galaxy cluster, but then after running tt_metal workload it stops working. Effectivelly UMD's read_from_non_mmio stops working.

We need to deep dive into remote communication to properly understand what is happening here. Vuk's asumption is that it might be the case that right after reset, default eth core kernels know how to communicate with each other so you can send a read request to any eth core and it will be resolved properly, but metal uses inactive eth cores for its workload, and then this functionality breaks.

**Comment by broskoTT** [comment]
We do plan on redesigning this part of UMDs API anyway, to relieve the burden of proper configuration on tt-metal. But this scenario still needs to be verified/investigated and understood what is exactly happening.

**Comment by broskoTT** [comment]
@tt-vjovanovic mentioned his experiment:
To get back on Galaxy after running Metal tests. We have hardcoded to use ETH cores that are being used on Galaxy (instead of ones used on N300) and it just works. So basically, if UMD would choose path to go directly to ETH core that is connected to device we want to talk with, we wouldn’t have a problem…

**Comment by broskoTT** [comment]
This will go away by itself at one point, no need to actively work on it.

**Comment by broskoTT** [comment]
This is still TBD. We're starting some work for tt-fabric to support multiprocessing better, this might pop up. This specific task is not currently high priority

**Comment by broskoTT** [comment]
This is more of a feature request than a true bug.
We can take a look at this once we refactor some of the stuff. Not a priority and not currently working on it.

**Comment by broskoTT** [comment]
Bojan Roško
  Wednesday at 11:28 AM
, a question regarding potential clash between umd and tt_metal on ETH FW. tt-fabric is out of context here.
Context is that we're trying to pinpoint all the potential issues that might come up from running multiple UMDs from multiple processes.
UMD uses active ETH links for communication with remote chips, during topology discovery for example. How would this mechanism work in parallel with some tt_metal CCL over the link? Does this work nicely or it doesn't have a chance on working together?
I know that current UMD code also tries to use IDLE eth links, which then reroute the communication through active ones, and tt_metal puts their own eth FW on IDLE ones, so that can break the mechanism. But this is easily solved, we can ignore this.


Umair Cheema
  Wednesday at 7:18 PM
Hi, ethernet firmware runs two tasks and context switches between them. One task is for handling UMD requests and second task runs tt-mtal/ccl kernel.
So on active ethernet cores, both can co-exist.
Idle ethernet cores are used by metal for dispatch. When that happens, that particular ethernet core cannot service umd requests. That is becasue we reset idle cores and load totally different runtime firmware that does not interface with UMD.


Umair Cheema
Ethernet port 15 is reserved for tools so, that port even if idle, it not repurposed for running dispatch kernels.


Bojan Roško
  Wednesday at 7:25 PM
Oh, that's good to know! So you're saying that we can potentially read arc telemetry from a remote chip, using UMD, while tt-metal workload is running?


Umair Cheema
yes, that's right.

**Comment by broskoTT** [comment]
This was fixed by some redesign in the meantime, and now that we're deprecating 4u is not of further interest anyway. Closing
